cobra2snooty.go

```diff
@@ -119,9 +121,22 @@ func GenDocs(cmd *cobra.Command, w io.Writer) error {
 	}
 	printOptions(buf, cmd)
 
-	if len(cmd.Example) > 0 {
+	// If it has an example but there's no #, print the example. If there's no example, don't print.
+	if len(examples) == 1 && len(cmd.Example) > 0 {
+		buf.WriteString(examplesHeader)
+		buf.WriteString(`.. code-block::
+`)
+		buf.WriteString(fmt.Sprintf("\n%s\n", indentString(cmd.Example, " ")))
+	}
+
+	// If it has an example with a #, print the header, then print each in a code block.
+	if len(examples) > 1 {
 		buf.WriteString(examplesHeader)
-		buf.WriteString(fmt.Sprintf("\n%s\n\n", indentString(cmd.Example, " ")))
+		for _, example := range examples[1:] {
+			buf.WriteString(`.. code-block::
+			`)
+			buf.WriteString(fmt.Sprintf("\n   #%s\n", indentString(example, " ")))
+		}
```

Comment:
FYI one small improvement here to reduce complexity

```go
	if cmd.Example != "" {
		// Create example substrings
		examples := strings.Split(cmd.Example, "# ")
		buf.WriteString(examplesHeader)
		if len(examples) == 1 { // If it has an example but there's no #, print the example. If there's no example, don't print.
			buf.WriteString(`.. code-block::
`)
			buf.WriteString(fmt.Sprintf("\n%s\n", indentString(cmd.Example, " ")))
		} else { // If it has an example with a #, print the header, then print each in a code block.
			for _, example := range examples[1:] {
				buf.WriteString(`.. code-block::
			`)
				buf.WriteString(fmt.Sprintf("\n   #%s\n", indentString(example, " ")))
			}
		}
	}
```
then anything inside the outside if can be made a function and remove the linting error